project.clj

```diff
@@ -1,3 +1,71 @@
+(def jruby-version "1.7.26")
+(def jffi-version "1.2.12")
+
+(def base-dependencies
+  "Base set of dependencies which are built into the project's artifact.
+  Note that these dependencies are customized somewhat when building an
+  uberjar.  See the `uberjar-dependencies` variable."
+  [['org.jruby/jruby-core jruby-version
+    :exclusions ['com.github.jnr/jffi
+                 'com.github.jnr/jnr-x86asm
+                 'org.ow2.asm/asm
+                 'org.ow2.asm/asm-analysis
+                 'org.ow2.asm/asm-commons
+                 'org.ow2.asm/asm-tree
+                 'org.ow2.asm/asm-util]]
+
+   ;; jruby-core has dependencies on discrete org.ow2.asm artifacts whereas
+   ;; other common Clojure projects like core.async declare a dependency on
+   ;; org.ow2.asm/asm-all, which provides a superset of the content of the
+   ;; discrete org.ow2.asm dependencies.  Defining asm-all here allows for
+   ;; conflict resolution to be possible in consuming projects.
+   ['org.ow2.asm/asm-all "5.0.3"]
+
+   ;; jffi and jnr-x86asm are explicit dependencies because, in JRuby's poms,
+   ;; they are defined using version ranges, and :pedantic? :abort won't
+   ;; tolerate this.
+   ['com.github.jnr/jffi jffi-version]
+   ['com.github.jnr/jffi jffi-version :classifier "native"]
+   ['com.github.jnr/jnr-x86asm "1.0.2"]
+   ['org.jruby/jruby-stdlib jruby-version]])
```

Comment:
For these you might be able to backtick the whole vector and unquote the version variables, like

```clojure
`[[org.jruby/jruby-core ~jruby-version
    :exclusions [com.github.jnr/jffi
                 com.github.jnr/jnr-x86asm
                 org.ow2.asm/asm
                 org.ow2.asm/asm-analysis
                 org.ow2.asm/asm-commons
                 org.ow2.asm/asm-tree
                 org.ow2.asm/asm-util]]
   [org.ow2.asm/asm-all "5.0.3"]
...
```

You can also do this for the sets below, except for the `joda-time` dependency. Using \` is safe on the other dependencies because \``[foo/bar]` expands to `[foo/bar]` while \``[bar]` expands to `[the.current.namespace/bar]`

But that set can be quoted with `'`. `'[bar]` expands to `[bar]`. We only need the backtick when we have variables that we need to unquote.

¯\\\_(ツ)\_/¯ maybe that's not a huge improvement